src/api_routes_websocket.rs

```diff
@@ -536,8 +537,11 @@ where
   Data: PerformApub + SendActivity<Response = <Data as PerformApub>::Response> + Send,
   for<'de> Data: Deserialize<'de>,
 {
-  let parsed_data: Data = serde_json::from_value(data)?;
-  let res = parsed_data.perform(&context, Some(id)).await?;
+  let parsed_data: WithAuth<Data> = serde_json::from_value(data)?;
+  let res = parsed_data
+    .data
+    .perform(&context, parsed_data.auth.clone(), Some(id))
+    .await?;
```

Comment:
See here for websocket, it wraps with a separate struct `WithAuth` which has an optional auth field. Similar to `WithContext` in federation code.